app/api/services/models/base.py

```diff
@@ -0,0 +1,12 @@
+# Base class for handling input params for PATCH endpoints.
+# Persist the original JSON input in _fields_set, so that
+# we can determine which fields were set by the user, and
+# which were omitted.
+class PatchParamsBase:
```

Comment:
I really like the intent to generalize and reuse the idea of PatchParams and get_set_params. In this case I don't think we need inheritance since we don't need polymorphism i.e. the functionality of `get_set_params` should be the same in all cases, and the only difference is in the "ApiModel" (in our case it's the "ApiUserModel").


I suggest we do something like

```
RequestModelType = TypeVar("RequestModelType", bound=BaseParams)
@dataclasses.dataclass
class PatchParams:
    params: RequestModelType
    fields_set: list[str]
```

then to use it 

```
class UserPatchParamsSchema(UserSchema):
    @marshmallow.post_load
    def make_user_patch_params(self, data: dict, **kwargs: dict[str, Any]) -> UserPatchParams:
        return PatchParams(
            params=UserParams(**kwargs),
            fields_set=list(kwargs.keys()), # just storing the keys feels less redundant to me
        )
```